installer/partitions.go

```diff
@@ -181,6 +181,16 @@ func (p *partitions) getAndCacheActivePartition(rootChecker func(system.StatComm
 			log.Debug("Setting active partition: ", mountCandidate)
 			return p.active, nil
 		}
+		if mountCandidate == p.rootfsPartA {
+			p.active = p.rootfsPartA
+			log.Debugf("Setting active partition from configuration and mount candidate: %s", p.active)
+			return p.active, nil
+		}
+		if mountCandidate == p.rootfsPartB {
+			p.active = p.rootfsPartB
+			log.Debugf("Setting active partition from configuration and mount candidate: %s", p.active)
+			return p.active, nil
+		}
```

Comment:
I *think* this should be ok, and yeah, the idea makes sense. This has previously proved to be quite a minefield though, so no promises until we run this through our full test suite.